core/src/main/java/org/bitcoinj/core/strategies/DenseThenSparseStrategy.java

```diff
@@ -0,0 +1,82 @@
+package org.bitcoinj.core.strategies;
+
+import com.google.common.base.Stopwatch;
+import org.bitcoinj.core.AbstractBlockChain;
+import org.bitcoinj.core.NetworkParameters;
+import org.bitcoinj.core.Sha256Hash;
+import org.bitcoinj.core.StoredBlock;
+import org.bitcoinj.store.BlockStore;
+import org.bitcoinj.store.BlockStoreException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Constructs a block locator using the strategy described on the <a href="https://en.bitcoin.it/wiki/Protocol_specification#getblocks">bitcoin wiki</a>
+ */
+public class DenseThenSparseStrategy implements BlockLocatorStrategy {
+
+    private static final Logger log = LoggerFactory.getLogger(DenseThenSparseStrategy.class);
+    private NetworkParameters params;
+
+    @Override
+    public List<Sha256Hash> createBlockLocator(AbstractBlockChain chain) {
+
+        //make a list of the block heights that we want to send
+        int height = chain.getBestChainHeight();
+        Set<Integer> indexes = new HashSet<Integer>();
+        int step = 1;
+        int minIndex = 0;
+        int maxDepth = getMaxDepth();
+        // stop once we reach a maximum depth
+        for (Integer index = height;index > 0 && (height - index) < maxDepth; index = index - step) {
```

Comment:
currently, the locator list will not go any deeper than the current implementation (100 deep).  Can be overridden in a sub class.